### PR TITLE
CDK phase 2: dual stack

### PR DESCRIPTION
**Pull request by tomrf1**
This step creates the new stack, and leaves the old stack in place.
It also creats a CNAME record which we will later use to route traffic to the new stack's loadbalancer.

### CODE REVIEW

File: cdk/lib/admin-console.ts [hunk 1]
```diff
@@ -1,15 +1,96 @@
...
+      monitoringConfiguration: {
+        noMonitoring: true,
+      },
+      userData,
+      roleConfiguration: {
+        additionalPolicies: policies,
+      },
+      scaling: { minimumInstances: 1, maximumInstances: 2 },
+      instanceType: InstanceType.of(InstanceClass.T4G, InstanceSize.MICRO),
+      withoutImdsv2: true,
```
Comment (tomrf1) [inline cdk/lib/admin-console.ts:74]: necessary for cloudwatch logging, see https://github.com/guardian/support-frontend/commit/76a015bc2461af10295c7ddb3dd0651ee92c23ae

File: cdk/lib/admin-console.test.ts [hunk 2]
```diff
@@ -5,7 +5,11 @@ import { AdminConsole } from './admin-console';
 describe('The AdminConsole stack', () => {
   it('matches the snapshot', () => {
     const app = new App();
-    const stack = new AdminConsole(app, 'AdminConsole', { stack: 'support', stage: 'TEST' });
+    const stack = new AdminConsole(app, 'AdminConsole', {
+      stack: 'support',
```
Comment (jacobwinch) [inline cdk/lib/admin-console.test.ts:9]: Minor thing (and personal preference really): I have started pulling my [`PROD` props into a const](https://github.com/guardian/amigo/blob/3f0bb6b1c75d2029eeddac0d6fcc27ee850c75e8/cdk/bin/cdk.ts#L20-L24) and [passing these into the test](https://github.com/guardian/amigo/blob/3f0bb6b1c75d2029eeddac0d6fcc27ee850c75e8/cdk/lib/amigo.test.ts#L9) as well as the [stack instantiation](https://github.com/guardian/amigo/blob/3f0bb6b1c75d2029eeddac0d6fcc27ee850c75e8/cdk/bin/cdk.ts#L26).

This helps to keep your snapshot as close as possible to the template that you will actually upload (i.e. the thing that you really care about not breaking!) - perhaps there are a few subtle bugs that can be caught this way as the number of stage-specific props grows.

File: cdk/lib/admin-console.ts [hunk 3]
```diff
@@ -1,15 +1,96 @@
...
+      }),
+      new GuPutS3ObjectsPolicy(this, 'PublicSettingsBucketPut', {
+        bucketName: 'gu-contributions-public',
+        paths: [`epic/${this.stage}/*`, `banner/${this.stage}/*`, `header/${this.stage}/*`],
+      }),
+    ];
+
+    const ec2App = new GuEc2App(this, {
+      applicationPort: 9000,
+      app: 'admin-console',
```
Comment (jacobwinch) [inline cdk/lib/admin-console.ts:60]: Minor: I think you can use your `app` const (from line 28) here

File: cdk/lib/admin-console.ts [hunk 4]
```diff
@@ -1,15 +1,96 @@
...
+      access: { scope: AccessScope.PUBLIC },
```
Comment (jacobwinch) [inline cdk/lib/admin-console.ts:61]: This feels a little odd for an internal tool, but it looks like the old load balancer is also configured in this way?

File: cdk/lib/admin-console.ts [hunk 5]
```diff
@@ -1,15 +1,96 @@
...
+    const template = new CfnInclude(this, 'YamlTemplate', {
       templateFile: yamlTemplateFilePath,
     });
+
+    // TODO Migrate to new LB
+    const oldElb = template.getResource('ElasticLoadBalancer') as CfnLoadBalancer;
+    new GuCname(this, 'cname', {
+      app,
+      domainName,
+      ttl: Duration.minutes(60),
```
[the remark this answers is not in the document]
Comment (tomrf1) [reply cdk/lib/admin-console.ts:92]: yep that's right!

File: cdk/lib/admin-console.ts (hunk 4 above)
Comment (tomrf1) [reply cdk/lib/admin-console.ts:61]: yes it actually used to be restricted, but was changed [here](https://github.com/guardian/support-admin-console/pull/118)!

**Pull request merged**